sklearn/utils/_array_api.py

```diff
@@ -1108,3 +1108,20 @@ def _tolist(array, xp=None):
         return array.tolist()
     array_np = _convert_to_numpy(array, xp=xp)
     return [element.item() for element in array_np]
+
+
+def _nan_to_num(array, xp=None):
+    """Substitutes NaN values of an array with 0 and inf values with the maximum or
+    minimum numbers available for the dtype respectively; like np.nan_to_num."""
+    xp, _ = get_namespace(array, xp=xp)
+    try:
+        array = xp.nan_to_num(array)
+    except AttributeError:  # currently catching exceptions from array_api_strict
+        array[xp.isnan(array)] = 0
+        if xp.isdtype(array.dtype, "real floating"):
+            array[xp.isinf(array) & (array > 0)] = xp.finfo(array.dtype).max
+            array[xp.isinf(array) & (array < 0)] = xp.finfo(array.dtype).min
+        else:  # xp.isdtype(array.dtype, "integral")
+            array[xp.isinf(array) & (array > 0)] = xp.iinfo(array.dtype).max
+            array[xp.isinf(array) & (array < 0)] = xp.iinfo(array.dtype).min
+    return array
```

Comment:
`_nan_to_num` is a leftover from the orginal branch and not needed for this PR, since here, we are converting to numpy arrays and `_nan_to_num` is only needed to work with `array_api_strict` arrays.

But we're using `np.nan_to_num` also in `silhouette_samples`, in `BaseSearchCV._format_results._store()` and in two tests. So I am not sure if it can stay.

Reply:
Oh, just seen [this](https://github.com/scikit-learn/scikit-learn/pull/30440#discussion_r1897933229). Maybe I have to remove it then, depending on what will be discussed on the issue.